Framework/API/src/ExperimentInfo.cpp

```diff
@@ -256,10 +257,12 @@ makeDetectorInfo(const Instrument &oldInstr, const Instrument &newInstr) {
  * @param visitor : Component visitor to query. Visitor MUST have been filled
  * via registerContents.
  */
-void ExperimentInfo::makeAPIComponentInfo(const InfoComponentVisitor &visitor) {
+void ExperimentInfo::makeAPIComponentInfo(
+    const InfoComponentVisitor &visitor,
+    boost::shared_ptr<Beamline::DetectorInfo> detectorInfo) {
 
   // Internal Beamline ComponentInfo
-  m_componentInfo = visitor.componentInfo();
+  m_componentInfo = visitor.componentInfo(detectorInfo);
```

Comment:
Yes, this is absolutely the plan. That would give the visitor just two relevant methods `componentInfo()` and `detectorInfo()`. But that's too large of a change to incorporate into this, so as you say, follow up.